### PR TITLE
Stocks (Lab3)

### PR DESCRIPTION
**Pull request by anton-cheg**

### CODE REVIEW

File: App.Stocks/Exceptions/IncorrectDateFormat.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace App.Stocks.Exceptions
+{
+    public class IncorrectParamsFormat : Exception
```
Comment (anton-cheg) [inline App.Stocks/Exceptions/IncorrectDateFormat.cs]: exception classes should end with "Exception"

File: App.Stocks/Filters/StockExceptionFilter.cs [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            switch (context.Exception)
+            {
+                case EntityNotExist entityNotExist:
+                    {
+                        context.HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
+
+                        int entityId = 0;
+                        var entityType = entityNotExist.EntityType;
+
+                        if (entityType.Equals(typeof(Company)))
```
Comment (anton-cheg) [inline App.Stocks/Filters/StockExceptionFilter.cs]: entity not found exception should contain info about identifier, used to search this type of entity

File: App.Stocks/Filters/StockExceptionFilter.cs [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+                        {
+                            entityId = GetStockIdParamFromPathRequest(context);
+                        }
+
+                        logger.LogWarning($"Type : {entityNotExist.EntityType.AssemblyQualifiedName}, Id {entityId}. Method: {entityNotExist.TargetSite}.");
+
+                        await context.HttpContext.Response.WriteAsync($"Entity with id : {entityId} and type {entityNotExist.EntityType.AssemblyQualifiedName} not found!");
+                        break;
+                    }
+                case IncorrectStockDate incorrectStockDate:
```
Comment (anton-cheg) [inline App.Stocks/Filters/StockExceptionFilter.cs]: each 'case' should correspind to one status code returned. 404 - EntityNotExistException. 400 - IncorrectParameterException. Too many case statements for the same or similar exception types

File: App.Stocks/Filters/StockExceptionFilter.cs [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+                    {
+                        context.HttpContext.Response.StatusCode = (int)HttpStatusCode.Forbidden;
+                        var companyId = GetCompanyIdParamFromPathRequest(context);
+                        logger.LogWarning(сompanyStocksIsPrivate, $"CompanyId: {companyId}. Method: {сompanyStocksIsPrivate.TargetSite}");
+                        await context.HttpContext.Response.WriteAsync($"You cannot view information about these stocks.Company {сompanyStocksIsPrivate.CompanyName} has restricted access to stock information.");
+                        break;
+                    }
+                default:
+                    {
+                        logger.LogWarning($"Method: {context.Exception.TargetSite}.");
```
Comment (anton-cheg) [inline App.Stocks/Filters/StockExceptionFilter.cs]: logWarning => logError

**Review by ghost — CHANGES_REQUESTED** [review]
ожидаю внесения оговоренных изменений

**Comment by anton-cheg** [comment]
> ожидаю внесения оговоренных изменений

Правки внесены

**Review by ghost — APPROVED** [review]

**Pull request merged**